pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -243,22 +244,27 @@ func (d *DependenciesDistributor) Reconcile(ctx context.Context, request reconci
 	bindingObject := &workv1alpha2.ResourceBinding{}
 	err := d.Client.Get(ctx, request.NamespacedName, bindingObject)
 	if err != nil {
+		// The resource may no longer exist, in which case we stop processing.
 		if apierrors.IsNotFound(err) {
-			klog.V(4).Infof("ResourceBinding(%s) has been removed.", request.NamespacedName)
-			return reconcile.Result{}, d.handleIndependentBindingDeletion(request.Namespace, request.Name)
+			return reconcile.Result{}, nil
 		}
 		klog.Errorf("Failed to get ResourceBinding(%s): %v", request.NamespacedName, err)
 		return reconcile.Result{}, err
 	}
 
 	// in case users set PropagateDeps field from "true" to "false"
 	if !bindingObject.Spec.PropagateDeps || !bindingObject.DeletionTimestamp.IsZero() {
-		return reconcile.Result{}, d.handleIndependentBindingDeletion(request.Namespace, request.Name)
+		err = d.handleIndependentBindingDeletion(request.Namespace, request.Name)
+		if err != nil {
+			klog.Errorf("Failed to cleanup attached bindings when delete independent binding(%s): %v", request.NamespacedName, err)
```

Comment:
``` suggestion
			klog.Errorf("Failed to cleanup attached bindings for independent binding(%s): %v", request.NamespacedName, err)
```
The ResourceBinding is probably not being deleted.

Reply:
Updated, thx~